packages/backend/src/utils/QueryBuilder/queryBuilder.ts

```diff
@@ -765,66 +765,54 @@ export class MetricQueryBuilder {
         const sqlFrom = this.getBaseTableFromSQL();
         const sqlLimit = this.getLimitSQL();
         const { sqlOrderBy, requiresQueryInCTE } = this.getSortSQL();
+        const ctes = [...dimensionsSQL.ctes];
+        let finalSelectParts: Array<string | undefined> = [
```

Comment:
Updated logic to have a single return. 
When it needs a new query, it pushes the `finalSelectParts` into the `ctes` and overrides the `finalSelectParts` with new parts. This may happen a few times.